Submodule JUCE contains modified content
diff --git a/JUCE/modules/juce_gui_basics/native/juce_mac_Windowing.mm b/JUCE/modules/juce_gui_basics/native/juce_mac_Windowing.mm
index 99893aaa0..5421916f1 100644
--- a/JUCE/modules/juce_gui_basics/native/juce_mac_Windowing.mm
+++ b/JUCE/modules/juce_gui_basics/native/juce_mac_Windowing.mm
@@ -613,23 +613,23 @@ static Image createNSWindowSnapshot (NSWindow* nsWindow)
 {
     JUCE_AUTORELEASEPOOL
     {
-        CGImageRef screenShot = CGWindowListCreateImage (CGRectNull,
-                                                         kCGWindowListOptionIncludingWindow,
-                                                         (CGWindowID) [nsWindow windowNumber],
-                                                         kCGWindowImageBoundsIgnoreFraming);
-
-        NSBitmapImageRep* bitmapRep = [[NSBitmapImageRep alloc] initWithCGImage: screenShot];
-
-        Image result (Image::ARGB, (int) [bitmapRep size].width, (int) [bitmapRep size].height, true);
-
-        selectImageForDrawing (result);
-        [bitmapRep drawAtPoint: NSMakePoint (0, 0)];
-        releaseImageAfterDrawing();
-
-        [bitmapRep release];
-        CGImageRelease (screenShot);
-
-        return result;
+        // CGImageRef screenShot = CGWindowListCreateImage (CGRectNull,
+        //                                                  kCGWindowListOptionIncludingWindow,
+        //                                                  (CGWindowID) [nsWindow windowNumber],
+        //                                                  kCGWindowImageBoundsIgnoreFraming);
+
+        // NSBitmapImageRep* bitmapRep = [[NSBitmapImageRep alloc] initWithCGImage: screenShot];
+
+        // Image result (Image::ARGB, (int) [bitmapRep size].width, (int) [bitmapRep size].height, true);
+
+        // selectImageForDrawing (result);
+        // [bitmapRep drawAtPoint: NSMakePoint (0, 0)];
+        // releaseImageAfterDrawing();
+
+        // [bitmapRep release];
+        // CGImageRelease (screenShot);
+
+        return {};
     }
 }
 

