diff --git a/JUCE/modules/juce_audio_plugin_client/Standalone/juce_StandaloneFilterApp.cpp b/JUCE/modules/juce_audio_plugin_client/Standalone/juce_StandaloneFilterApp.cpp
index cb099283..9879cd8a 100644
--- a/JUCE/modules/juce_audio_plugin_client/Standalone/juce_StandaloneFilterApp.cpp
+++ b/JUCE/modules/juce_audio_plugin_client/Standalone/juce_StandaloneFilterApp.cpp
@@ -67,7 +67,12 @@ public:
         appProperties.setStorageParameters (options);
     }
 
-    const String getApplicationName() override              { return JucePlugin_Name; }
+    const String getApplicationName() override              
+    { 
+        String pluginString =  File::getSpecialLocation(File::currentExecutableFile).getFileNameWithoutExtension();
+        return pluginString; 
+     }
+     
     const String getApplicationVersion() override           { return JucePlugin_VersionString; }
     bool moreThanOneInstanceAllowed() override              { return true; }
     void anotherInstanceStarted (const String&) override    {}
